### ISSUE TITLE
linter: react/invariant Ref type environment did not converge

### ISSUE THREAD

**Issue by pcorpet** [oxc-project/oxc#26181]
### What version of Oxlint are you using?

1.80.0

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "categories": {
    "correctness": "off"
  },
  "plugins": ["react"],
  "rules": { "react/invariant": "error" }
}
```

### What happened?

Ref type environment did not converge.

When running oxlint, I get a message to report this error here: `Please report this internal React Compiler error to Oxc with a minimal reproduction`

```ts
import { useMemo, useRef } from 'react'

export interface ICancellableDebounceFn<TFunc extends (...args: any) => void> {
	cancel: () => void
	(...args: Parameters<TFunc>): void
}

const useCancellableDebounce = <TFunc extends (...args: any) => void>(
	fn: TFunc,
	time = 100,
): ICancellableDebounceFn<TFunc> => {
	const timeoutRef = useRef<NodeJS.Timeout | null>(null)

	return useMemo(() => {
		const cancelDebounce = () => {
			if (timeoutRef.current) {
				clearTimeout(timeoutRef.current)
			}
		}

		const debouncedFn = (...args: Parameters<TFunc>) => {
			cancelDebounce()
			timeoutRef.current = setTimeout(() => fn(...args), time)
		}

		debouncedFn.cancel = cancelDebounce
		return debouncedFn
	}, [fn, time])
}

export { useCancellableDebounce }
```
